### PR TITLE
Show info tooltip from cards on dashboards

### PR DESCRIPTION
**Pull request by tanzoniteblack**
Addresses https://github.com/metabase/metabase/issues/3980

screen shot 2017-01-24 at 10 28 31 am

###### TODO 
-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

### CODE REVIEW

**Comment by tanzoniteblack** [comment]
Closing this because the tool tip shows up multiple times on some charts with multiple headers, will try to fix and reopen

**Comment by tanzoniteblack** [comment]
Ok, fixed bug (see attached photo for what tool tips look like on graphs)

screen shot 2017-01-24 at 1 05 54 pm

**Comment by tlrobinson** [comment]
@kdoh What were your thoughts on the design for this?

**Comment by kdoh** [comment]
@tlrobinson / @tanzoniteblack  I think the main consideration here is that with a lot of cards with a lot of descriptions comes a lot of info dots and therefore a lot of potential visual noise. I'd be more inclined to just show the description tooltip icon if you're hovered on a particular card.

**Comment by kdoh** [comment]
Hey @tanzoniteblack. Do you need any more guidance to move this forward? I'm happy to provide a quick design of the solution I described above if that would help?

**Comment by tanzoniteblack** [comment]
I didn't actually catch that you had responded to this originally :)

To make sure I understand, @kdoh , you like the idea, but you want to change the design so that the `i` icon only appears when you're hovering over the card? Otherwise everything's the same, with having the actual text only appear over hover of the `i`?

**Comment by kdoh** [comment]
@tanzoniteblack Yup that's exactly right.

**Comment by tanzoniteblack** [comment]
That should be easy enough to do with just the CSS; just need to fix the merge conflicts first. Should have an update on this soon.

File: frontend/src/metabase/visualizations/components/LegendHeader.jsx
Comment (tlrobinson) [inline frontend/src/metabase/visualizations/components/LegendHeader.jsx]: Could you revert the indentation of this file? It's hard to see exactly what changed.

Comment (tanzoniteblack) [inline frontend/src/metabase/visualizations/components/LegendHeader.jsx]: Sorry about that, will also close this for the moment so you stop getting updates while I'm working on it and it's more obvious when I'm actually done

**Comment by tlrobinson** [comment]
@tanzoniteblack We have a similar behavior for the action buttons on the dashboard cards, using the `.DashCard-actions` selector, however those target the `.DashCard` selector, which is only used on dashboards. To ensure we get the same behavior in both dashboards and elsewhere visualizations are rendered we might want to add a `Visualization` class on the root of the `Visualization.jsx` component, then something like:

```css
.Visualization .Visualization-description {
  opacity: 0;
  transition: opacity .3s linear;
}

.Visualization:hover .Visualization-description {
  height: initial;
  opacity: 1;
}
```

@kdoh does that seem about right?

**Comment by tanzoniteblack** [comment]
Sorry about the indentation issues in the merge request. I have not quite figured out to get my emacs config to align with whatever indentation scheme is being used exactly.

**Comment by kdoh** [comment]
@tanzoniteblack No worries. We use [editorconfig](http://editorconfig.org/). [This emacs plugin](https://github.com/editorconfig/editorconfig-emacs) might help.

**Comment by tanzoniteblack** [comment]
Indentation issues aside (I'll definitely check out editorconfig and see if that fixes this going forward), this branch now will show info tip icon only on card hover and the info itself only on hover over that icon.

**Comment by camsaul** [comment]
@tanzoniteblack FWIW I use Emacs with the editorconfig package and it works for me

**Review by kdoh — CHANGES_REQUESTED** [review]
Hey @tanzoniteblack. Thanks again for all this. A couple quick bits of feedback.

Rather than add new CSS I'd prefer to use our existing hover utilities on this. Here's how that works:

On line 116 of DashCard.jsx you can add the classes `hover-parent hover--visibility`

Then in `LegendItem.jsx` the following change gives us the same behavior you have currently with no additional CSS required.
```jsx
                  { description &&
                      <div className="hover-child">
                          <Tooltip tooltip={description} maxWidth={'22em'}>
                              <Icon name='info' />
                          </Tooltip>
                      </div>
                  }
```

One other thing is that this update doesn't quite take into account scalar dash cards. I think all that's required is to add the same icon and functionality to that template since scalar cards don't use the same header that other dash cards do.

Let me know if that makes sense.

**Comment by tanzoniteblack** [comment]
@kdoh makes sense and implemented. Also added support for scalar dashboard cards

**Review by kdoh — APPROVED** [review]

**Comment by kdoh** [comment]
@tanzoniteblack Awesome! I'm on a plane and about to land but I'll give it another look when I'm on the ground wth wifi.

**Comment by kdoh** [comment]
This looks good! Thanks @tanzoniteblack.

**Comment by tlrobinson** [comment]
Note that this will only work on dashboards since the class is on DashCard rather than Visualization. Not sure that matters at this point, since we don't display the title in the query builder, except in chart settings.

**Comment by tanzoniteblack** [comment]
@tlrobinson where else would we expect this to work besides dashboards? I've only ever interacted with questions from the question view (where there's already an info icon for the description) and from dashboards (which this PR adds).

Is there a 3rd place that visualizations are shown at?

**Comment by tanzoniteblack** [comment]
@kdoh There's nothing left waiting on me to do before this gets merged, right?

**Comment by kdoh** [comment]
@tanzoniteblack Nope! Thanks for your contribution.

**Pull request merged**
